examples/sampling/pyg/node_classification.py

```diff
@@ -75,104 +80,100 @@ def __init__(self, in_size, hidden_size, out_size):
         self.layers.append(SAGEConv(hidden_size, hidden_size))
         self.layers.append(SAGEConv(hidden_size, out_size))
 
-    def forward(self, blocks, x, device):
-        h = x
-        for i, (layer, block) in enumerate(zip(self.layers, blocks)):
-            src, dst = block.edges()
-            edge_index = torch.stack([src, dst], dim=0)
-            h_src, h_dst = h, h[: block.number_of_dst_nodes()]
-            h = layer((h_src, h_dst), edge_index)
-            if i != len(blocks) - 1:
-                h = F.relu(h)
-        return h
+    def forward(self, x, edge_index):
+        for i, layer in enumerate(self.layers):
+            x = layer(x, edge_index)
+            if i != len(self.layers) - 1:
+                x = x.relu()
+                x = F.dropout(x, p=0.5, training=self.training)
+        return x
 
+    def inference(self, args, dataloader, x_all, device):
+        """Conduct layer-wise inference to get all the node embeddings."""
+        for i, layer in tqdm(enumerate(self.layers), "inference"):
+            xs = []
+            for minibatch in dataloader:
+                # Call `to_pyg_data` to convert GB Minibatch to PyG Data.
+                pyg_data = minibatch.to_pyg_data()
+                n_ids = minibatch.node_ids().to("cpu")
+                x = x_all[n_ids].to(device)
```

Comment:
```suggestion
                x = gb.index_select(x_all, minibatch.node_ids())
```
Works from GPU when `x_all` is pinned and node ids are on GPU.

Reply:
However, in our case, because we use the read function, `x_all` is probably on the GPU?

Reply:
Here we hope to use the same inference function as the PyG example with least modification. In their example, `x_all` is on the CPU

Reply:
Got it. We might want to get the initial `x_all` in a different way then because it moves the whole feature tensor to the GPU by default when features are pinned.

Reply:
We might want to directly access the stored tensor.

Reply:
I think we can put this for later since it's not a blocker.